### PR TITLE
made table for badges and added doxy info

### PR DESCRIPTION
**Pull request by SteveMacenski**
Updated readme file.

@ruffsl I noticed on DockerHub the docker file it says its using is (https://hub.docker.com/r/rosplanning/navigation2/dockerfile) the dummy file, though I see from the `.dockerhub` I believe it shows it using the `Dockerfile` which is the "real" one. Can you confirm for me what its actually using and can we get rid of the dummy file?

### CODE REVIEW

**Comment by ruffsl** [comment]
> Can you confirm for me what its actually using and can we get rid of the dummy file?

So the Docker Hub build setup is a little peculiar given that the `FROM` directive for the parent image is specified using a `ARG` variable.

https://github.com/ros-planning/navigation2/blob/1ec7b205ddee240273ac6ba5d218d050e8ab4f4c/Dockerfile#L9-L12

This allows us to specify the target parent image at build time, e.g:

```
docker build --build-arg FROM_IMAGE=ros:eloquent ...
```

The caveat is that this also interferes with Docker Hub's internal introspection to determine what Docker Hub repo to trigger build from when the automated repo option is enabled. As a work around, we have the Docker Hub point to the dummy Dockerfile that builds and triggers `FROM` the intended parent repo, while the build hook override uses the symlink Dockerfile (pointing to the root directory) with the appropriate build args.

https://github.com/ros-planning/navigation2/blob/1ec7b205ddee240273ac6ba5d218d050e8ab4f4c/.dockerhub/debug/hooks/build#L4-L11

Thus, we avoid having to duplicate any Dockerfiles that could otherwise fall out of sync, while retaining the automated build and triggering from parent image repos. An inconsistency is that Dockerhub web UI displays the dummy Dockerfile, but that UI isn't too useful anyway given it only shows one Dockerfile per repo, even if the repo has more than one or multiple tags.

 In short, the REAME should direct the user to just use root folder Dockerfile, all the stuff in `.dockerhub` folder is irrelevant to most users.

**Comment by SteveMacenski** [comment]
Got it, so I should just ignore that on the dockerhub interface.

File: README.md [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+# Navigation2 Stack
 
-ROS2 Navigation System
 
-[![Build Status](https://circleci.com/gh/ros-planning/navigation2/tree/master.svg?style=svg)](https://circleci.com/gh/ros-planning/navigation2/tree/master) CircleCI
+| Builder                | Status                    |
+|----------------------|-------------------------|
+| CircleCI             | [![Build Status](https://circleci.com/gh/ros-planning/navigation2/tree/master.svg?style=svg)](https://circleci.com/gh/ros-planning/navigation2/tree/master)   |
+| DockerHub       | [![Build Status](https://img.shields.io/docker/cloud/build/rosplanning/navigation2.svg?label=build)](https://hub.docker.com/r/rosplanning/navigation2)       |
+| ROS2 Build Farm   | [![Build Status](http://build.ros2.org/job/Cdev__navigation2__ubuntu_bionic_amd64/badge/icon)](http://build.ros2.org/job/Cdev__navigation2__ubuntu_bionic_amd64/)      |
```
Comment (mkhansen-intel) [inline README.md]: This link appears to be broken, even on the existing README, any idea why?

Comment (mkhansen-intel) [inline README.md]: The ROS2 Build Farm link appears to be broken, even on the prior version, can you fix that?

File: README.md [hunk 2]
```diff
@@ -18,6 +17,8 @@ The ROS 2 Navigation System is the control system that enables a robot to autono
 # Documentation
 For detailed instructions on how to install and run the examples, please visit our [documentation site](https://ros-planning.github.io/navigation2/).
 
+Run `doxygen` in the root of this repository. It will generate a `/doc/*` directory containing the documentation. Entrypoint in a browser is index.html.
```
Comment (mkhansen-intel) [inline README.md]: This is confusing to me, doxygen is different than the github pages documentation, but this makes it seem like doxygen is how that documentation is generated. I think a sub-heading like 'API documentation' or something would be good to clarify the difference.

File: README.md (hunk 1 above)
Comment (SteveMacenski) [reply README.md]: Its not usually broken, build farm isnt having a good time right now if you check your emails. You can safely ignore it. It was working yesterday

Comment (SteveMacenski) [reply README.md]: Its not usually broken, build farm isnt having a good time right now if you check your emails. You can safely ignore it. It was working yesterday

File: README.md (hunk 2 above)
Comment (SteveMacenski) [reply README.md]: @mkhansen-intel good point - done.
